doc/whats-new.md

```diff
@@ -1,358 +1,387 @@
+
+In both cases the engine raises warnings asking the user to take
+action if problems are identified. Both changes were motivated by the request of
+making the engine less magic and giving more control to the user. They
+are fully documented here
+
+https://github.com/gem/oq-engine/blob/engine-3.9/doc/adv-manual/effective-realizations.html
+
+and here:
+
+https://github.com/gem/oq-engine/blob/engine-3.9/doc/adv-manual/common-mistakes.rst#pointsource_distance
+
+# Logic trees
+
+Most of the work on this release went into a deep refactoring of the
+logic tree code. From the user perspective the most notable change is
+that the time needed to parse and manage the source models has been
+substantially reduced. This is particularly visible in the case of the
+complex logic trees used for power plant hazard analysis and similar
+studies (we are talking about orders of magnitude speedups). For continental
+scale calculations the speedup is very sensible when running in preclassical
+mode or in single site calculations, while it is insignificant compared
+to the total runtime in the other cases.
+
+Internally, the basic logic tree classes, as well as the code to
+manage the uncertainties, have been moved into hazardlib. The change
+makes it possible for a power user to introduce new custom
+uncertainties with a little Python coding, while before
+adding new uncertainties was extremely difficult, even for a core
+developer. Users with an interest on such topics should
+contact us and we can give some guidance.
+
+The removal of the *automatic* reduction of the GMPE logic tree
+feature allowed substantial simplifications and made it possible to
+infer in advance the size of a calculation, thus giving early warnings
+in the case of calculations too big to run. It is still possible to
+reduce the GMPE logic tree *manually*, by discarding irrelevant
+tectonic region types (a TRT is irrelevant if there are no sources for
+that TRT within the integration distance). The engine will tell you
+automatically which are the irrelevant TRTs, even without running a
+full calculation.
+
+There is a new (and much requested) feature, the ability to add sources
+to a source model as a special kind of uncertainty.  The feature is
+called `extendModel` and is documented here:
+
+https://github.com/gem/oq-engine/blob/engine-3.9/doc/adv-manual/special-features.rst#extendmodel
+
+Even if the engine does not offer any built-in way to plot logic trees, an
+example of how you can do it yourself by using the
+[ete3](http://etetoolkit.org/) library has been added.
+
+# New optimizations
+
+There were *lots* of optimizations and improvements.
+
+The most impressive optimization is the enhancement of the point
+source collapsing mechanism for *site specific classical
+calculations*.  This can easily give an order of magnitude speedup for
+calculations dominated by point sources, i.e. most calculations. The
+price to pay is a small reduction in precision, as discussed here:
```

Comment:
"a small reduction in precision, as discussed here" would replace it with "a different behaviour for a feature that maintains the same naming. Additional information is available here".

Reply:
No, this is a totally new feature not available before. I am not talking about the pointsource_distance approximation.